src/nemos/basis/_basis_mixin.py

```diff
@@ -48,9 +212,32 @@ def _compute_features(self, *xi: ArrayLike | Tsd | TsdFrame | TsdTensor):
         out = self._evaluate(*(np.reshape(x, (x.shape[0], -1)) for x in xi))
         return np.reshape(out, (out.shape[0], -1))
 
-    def set_kernel(self) -> "EvalBasisMixin":
+    def setup_basis(self, *xi: NDArray) -> Basis:
         """
-        Prepare or compute the convolutional kernel for the basis functions.
+        Set all basis states.
+
+        This method corresponds sklearn transformer ``fit``. As fit, it must receive the input and
+        it must set all basis states, i.e. ``kernel_`` and all the states relative to the input shape.
+        The difference between this method and the transformer ``fit`` is in the expected input structure,
+        where the transformer ``fit`` method requires the inputs to be concatenated in a 2D array, while here
+        each input is provided as a separate time series for each basis element.
+
+        Parameters
+        ----------
+        xi:
+            Input arrays.
+
+        Returns
+        -------
+        :
+            The basis with ready for evaluation.
+        """
+        self.set_input_shape(*xi)
+        return self
+
+    def _set_input_independent_states(self) -> "EvalBasisMixin":
```

Comment:
currently this only appears to call `set_kernel` in the convolutional mixin and does nothing anywhere else -- do we really need it?

Reply:
yes, and also for the future: I want to funnel every other setting through this method (for basis that do not exist yet :))